### PR TITLE
Feature parity for Hub Roles list

### PR DESCRIPTION
**Pull request by vidyanambiar**
- Filter roles 
    - Filter by name, editability (Editable/Built-in) and type (Galaxy-only/All roles)
    - The initial list is filtered on "Galaxy-only" roles, the filter dropdown has an "All roles" option to view all roles.
- A deviation from the current UI is to make the role name a link for consistency with other list as well as the Role details for EDA. Edit/delete will be available on that page too (the details page will be added separately)
- Deletion of roles from the list
- Tests

### Existing UI:
Screenshot 2023-11-29 at 10 44 43 AM

### New UI:
image

https://github.com/ansible/ansible-ui/assets/43621546/62589b77-7d26-4e7b-af81-afdd3d52771a

### CODE REVIEW

File: cypress/support/hub-commands.ts [hunk 1]
```diff
@@ -52,7 +58,7 @@ Cypress.Commands.add(
       cy.wait(300);
 
       cy.requestGet<HubItemsResponse<CollectionVersionSearch>>(
-        hubAPI`/v3/plugin/ansible/search/collection-versions/?namespace=${namespaceName}&name=${collectionName}`
+        `${apiPrefix}/v3/plugin/ansible/search/collection-versions/?namespace=${namespaceName}&name=${collectionName}`
       ).then((itemsResponse) => {
```
Comment (nixocio) [inline cypress/support/hub-commands.ts:62]: This one is an intentional change?

Comment (vidyanambiar) [reply cypress/support/hub-commands.ts:62]: Nope, I'll remove it. 👍

File: cypress/support/formatApiPathForHub.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    url += fragment;
+    if (index !== strings.length - 1) {
+      url += encodeURIComponent(`${values.shift() ?? ''}`);
+    }
+  });
+
+  return url;
+}
+
+export function hubAPI(strings: TemplateStringsArray, ...values: string[]) {
```
[the remark this answers is not in the document]
Comment (vidyanambiar) [reply cypress/support/formatApiPathForHub.ts:29]: My inclination would be to keep the function's signature the same across the frontend code and tests for ease of use and a consistent dev experience. WDYT?

[the remark this answers is not in the document]
Comment (vidyanambiar) [reply cypress/support/formatApiPathForHub.ts:29]: I agree that it helps with reducing the duplication but I'm concerned that changing the call signature will introduce unnecessary inconsistency in how we're used to calling those utils, just for tests. WDYT?

[the remark this answers is not in the document]
Comment (nixocio) [reply cypress/support/formatApiPathForHub.ts:29]: Yeah, lets change just the min necessary, good point.

**Pull request merged**